File: src/main/java/me/aov/sellgui/api/SellGUIPriceProvider.java
package me.aov.sellgui.api;

import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public interface SellGUIPriceProvider {
    String getName();

    default int getPriority() {
        return 0;
    }

    default boolean isAvailable() {
        return true;
    }

    double getSellPrice(Player player, ItemStack itemStack);

    default void onItemsSold(Player player, List<SoldItem> soldItems, double totalPrice) {
    }
}

File: src/main/kotlin/me/aov/sellgui/api/SellGUIPriceProvider.kt
package me.aov.sellgui.api

import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

interface SellGUIPriceProvider {
    fun getName(): String

    fun getPriority(): Int = 0

    fun isAvailable(): Boolean = true

    fun getSellPrice(player: Player, itemStack: ItemStack): Double

    fun onItemsSold(player: Player, soldItems: List<SoldItem>, totalPrice: Double) = Unit
}
